mautic/mautic
# New maintenance CLI command to purge old data

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | Y
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  |  n/a
| Developer Documentation PR | https://github.com/mautic/developer-documentation/pull/17
| User Documentation PR | https://github.com/mautic/documentation/pull/62

## Description

This adds a new command that uses an event to purge old data. Currently audit log, visitors (anonymous leads) and visitor page hits are supported. 

## Steps to test this PR

1) BACKUP YOUR DATABASE :-)

2) Run the command `php app/console mautic:maintenance:cleanup --days-old=365 --dry-run`

It should display a table with the number of records that will get deleted. You may need to adjust the `--days-old` value accordingly to test against your data.

3) Run the command again without `--dry-run`. You should be prompted to confirm. Entering `y` should result in a table of deleted results. 

4) VERIFY that the deleted results were appropriate:
leads - last_activity older than given `days-old` and date_identified is null
page_hits - page hits for visitors (leads) with last_activity older than given days old and date_identified is null
audit_log - entries with date_added older than given `days-old`
